lib/iris/fileformats/pp.py

```diff
@@ -1955,23 +1955,33 @@ def _convert_constraints(constraints):
     constraints = iris._constraints.list_of_constraints(constraints)
     pp_constraints = {}
     unhandled_constraints = False
+
+    def make_func(stashobj):
```

Comment:
please may you document this function just a little, mainly the intent for its use

it looks to me like this is a private function, only used once.  If so, I would prefer to see if named with a leading `_`, so `_make_func`

I wonder if you could just place this logic in line in the code, as it's only used once, I don't think it needs to be a function,